R/gi_fetch_templates.R

```diff
@@ -53,7 +53,7 @@ gi_fetch_templates <-
     template_name <- paste(template_name, collapse = ",")
 
     r <- curl::curl_fetch_memory(
-      glue::glue("https://www.gitignore.io/api/{template_name}")
+      glue::glue("https://www.toptal.com/developers/gitignore/api/{template_name}")
```

Comment:
Reusing the helper I suggest above, this line then becomes

```r
  glue::glue("{backend_url()}/api/{template_name}")
```

And you could follow the same idea to construct all the urls that I see changed in this PR.